libs/feature/map/src/lib/map-context/map-context.service.ts

```diff
@@ -54,46 +55,70 @@ export class MapContextService {
   }
 
   createLayer(layerModel: MapContextLayerModel): Layer {
-    const { type, url, urls, name } = layerModel
+    const { type } = layerModel
     const style = this.styleService.styles.default
     switch (type) {
       case MapContextLayerTypeEnum.XYZ:
         return new TileLayer({
           source: new XYZ({
-            url,
-            urls,
+            url: 'url' in layerModel ? layerModel.url : undefined,
+            urls: 'urls' in layerModel ? layerModel.urls : undefined,
           }),
         })
       case MapContextLayerTypeEnum.WMS:
         return new TileLayer({
           source: new TileWMS({
-            url,
-            params: { LAYERS: name },
+            url: layerModel.url,
+            params: { LAYERS: layerModel.name },
           }),
         })
       case MapContextLayerTypeEnum.WFS:
         return new VectorLayer({
           source: new VectorSource({
             format: new GeoJSON(),
             url: function (extent) {
-              return `${url}?service=WFS&version=1.1.0&request=GetFeature&outputFormat=application/json&typename=${name}&srsname=EPSG:3857&bbox=${extent.join(
-                ','
-              )},EPSG:3857`
+              return `${
```

Comment:
could have kept the dedicated extractions of var in the case statement, more readable IMO, not a big deal.

Reply:
this was actually not working with the stricter types, since some properties might not be present (e.g. for geojson it's either url or data, not both)

Reply:
you can do the residual extraction in the `case` statement, casting with `as` to the correct type.